pkg/controller/network/zone/controller.go

```diff
@@ -0,0 +1,202 @@
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-aws/apis/network/v1alpha3"
+	"github.com/crossplane/provider-aws/pkg/clients/zone"
+	"github.com/crossplane/provider-aws/pkg/controller/utils"
+)
+
+const (
+	errUnexpectedObject = "The managed resource is not an Zone resource"
+	errCreate           = "failed to create the Zone resource"
+	errDelete           = "failed to delete the Zone resource"
+	errUpdate           = "failed to update the Zone resource"
+	errGet              = "failed to get the Zone resource"
+	errKubeUpdate       = "failed to update the Zone custom resource"
+)
+
+// SetupZone adds a controller that reconciles Hosted Zones.
+func SetupZone(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha3.ZoneGroupKind)
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha3.Zone{}).
+		Complete(managed.NewReconciler(
+			mgr, resource.ManagedKind(v1alpha3.ZoneGroupVersionKind),
+			managed.WithExternalConnecter(
+				&connector{client: mgr.GetClient(),
+					newClientFn: zone.NewClient,
+					awsConfigFn: utils.RetrieveAwsConfigFromProvider},
+			),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(
+				mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithInitializers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(
+				mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) zone.Client
+	awsConfigFn func(context.Context, client.Reader, *corev1.ObjectReference) (*aws.Config, error)
+}
+
+func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha3.Zone)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+
+	awsconfig, err := conn.awsConfigFn(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	c := conn.newClientFn(awsconfig)
+
+	return &external{kube: conn.client, client: c}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client zone.Client
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha3.Zone)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	err := e.kube.Update(ctx, configSanitize(cr))
```

Comment:
Since the `comment` is the only field that can be updated, therefore I decided that if none is provided it best to set it to a default value so that it later can be checked against for the updation event. That is the `configSanitize` is supposed to check the current value and set it to its type default if nil. Which we then update the Kubernetes object with.